File: cpp/leetcode/393.cpp
class Solution {
public:
    bool validUtf8(vector<int>& data) {
        int cnts = 0;
        for(auto num : data){
            if(cnts == 0){
                if((num >> 5) == 0b110) cnts = 1;
                else if((num >> 4) == 0b1110) cnts = 2;
                else if((num >> 3) == 0b11110) cnts = 3;
                else if((num >> 7)) return false;
            } else {
                if((num >> 6) != 0b10) return false;
                cnts --;
            }
        }
        return cnts == 0;
    }
};

File: python/leetcode/393.py
class Solution(object):
    def validUtf8(self, data):
        cnts = 0
        for num in data:
            print num, num >> 5, num >> 4, num >> 3, num >> 7
            if not cnts:
                if (num >> 5) == 0b110:
                    cnts = 1
                elif (num >> 4) == 0b1110:
                    cnts = 2
                elif (num >> 3) == 0b11110:
                    cnts = 3
                elif (num >> 7):
                    return False
            else:
                if (num >> 6) != 0b10:
                    return False
                cnts -= 1
        return cnts == 0